kf_lib_data_ingest/etl/load/load.py

```diff
@@ -106,29 +99,52 @@ def __init__(
         self.uid_cache = defaultdict(dict)
         self.uid_cache_db = Sqlite3Worker(self.uid_cache_filepath)
 
-    def _validate_entities(self, entities_to_load):
+    def _validate_entities(self, entities_to_load, msg):
         """
         Validate that all entities in entities_to_load are one of the
-        target concepts specified in the target_api_config.target_concepts
+        target concepts specified in the target_api_config.all_targets
         """
-        assert_safe_type(entities_to_load, list)
-        assert_all_safe_type(entities_to_load, str)
-
+        target_names = [t.__name__ for t in self.target_api_config.all_targets]
+        lower_names = [n.lower() for n in target_names]
+        snake_names = [snake_case(name) for name in target_names]
         invalid_ents = [
             ent
             for ent in entities_to_load
-            if ent not in self.target_api_config.target_concepts
+            if (ent not in self.target_api_config.all_targets)
```

Comment:
default is already lowercase snake but I wanted to support different variants of the names. This will be obsolete anyway if adding a name attribute to the classes